File: app/src/main/java/com/agotakiss/androidtest/models/Movie.java
package com.agotakiss.androidtest.models;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

public class Movie implements Serializable {

    @SerializedName("poster_path")
    private String posterPath;

    @SerializedName("adult")
    private boolean isAdult;

    @SerializedName("overview")
    private String overview;

    @Override
    public String toString() {
        return "Movie{" +
                "posterPath='" + posterPath + '\'' +
                ", isAdult=" + isAdult +
                ", overview='" + overview + '\'' +
                ", releaseDateText='" + releaseDateText + '\'' +
                ", genreIdList=" + genreIdList +
                ", genres=" + genres +
                ", id=" + id +
                ", originalTitle='" + originalTitle + '\'' +
                ", originalLanguage='" + originalLanguage + '\'' +
                ", title='" + title + '\'' +
                ", backdropPath='" + backdropPath + '\'' +
                ", popularity=" + popularity +
                ", voteCount=" + voteCount +
                ", video=" + video +
                ", averageVote=" + averageVote +
                '}';
    }

    @SerializedName("release_date")
    private String releaseDateText;

    @SerializedName("genre_ids")
    private List<Integer> genreIdList;

    private List<Genre> genres;

    @SerializedName("id")
    private int id;

    @SerializedName("original_title")
    private String originalTitle;

    @SerializedName("original_language")
    private String originalLanguage;

    @SerializedName("title")
    private String title;

    @SerializedName("backdrop_path")
    private String backdropPath;

    @SerializedName("popularity")
    private float popularity;

    @SerializedName("vote_count")
    private int voteCount;

    @SerializedName("video")
    private boolean video;

    @SerializedName("vote_average")
    private float averageVote;

}

File: app/src/main/java/com/agotakiss/androidtest/domain/models/Movie.kt
package com.agotakiss.androidtest.domain.models

import java.io.Serializable

data class Movie(val posterPath: String? = null,
                 val isAdult: Boolean = false,
                 val overview: String? = null,
                 val releaseDateText: String? = null,
                 val genres: List<Genre>? = null,
                 val id: Int = 0,
                 val originalTitle: String? = null,
                 val originalLanguage: String? = null,
                 val title: String? = null,
                 val backdropPath: String? = null,
                 val popularity: Float = 0.toFloat(),
                 val voteCount: Int = 0,
                 val isVideo: Boolean = false,
                 val averageVote: Float = 0.toFloat()
):Serializable